crates/cairo-coverage/src/input/filter.rs

```diff
@@ -76,15 +86,23 @@ impl Filter {
             labels.insert(ItemLabel::Macro);
         }
 
-        // TODO(#55)
-        // TODO: We should probably filter by path to user project not by path to cache
-        // TODO: Can get this from source_sierra_path in call trace
-        if file_location.contains("com.swmansion.scarb") || file_location.contains(".cache/scarb") {
-            labels.insert(ItemLabel::NonUserFunction);
-        } else {
+        if file_location.contains(&self.user_project_path) {
             labels.insert(ItemLabel::UserFunction);
+        } else {
+            labels.insert(ItemLabel::NonUserFunction);
         }
 
         labels
     }
 }
+
+fn find_user_project_path(source_sierra_path: &str) -> Result<String> {
+    Utf8PathBuf::from(source_sierra_path)
+        .parent()
+        .filter(|parent| parent.ends_with(SNFORGE_SIERRA_DIR))
```

Comment:
how strict do you think should we be here? is `.ends_with` good enough? Maybe we should match against exact dir name instead?